framework/src/DevObj.cpp

```diff
@@ -165,18 +166,22 @@ int DevObj::start(void)
 		return -3;
 
 	} else {
-		WorkMgr::getWorkHandle(measure, this, m_sample_interval_usecs, m_work_handle);
+		do {
+			WorkMgr::getWorkHandle(measure, this, m_sample_interval_usecs, m_work_handle);
+
+			if (!m_work_handle.isValid()) {
+				return -m_work_handle.getError();
+			}
 
-		if (m_work_handle.isValid()) {
 			DF_LOG_DEBUG("DevObj::start schedule %s", m_name);
 			WorkMgr::schedule(m_work_handle);
 
-		} else {
-			return -4;
-		}
+			//if someone else calls getWorkHandle at the same time, both will return the same handle,
+			//and one of the schedule() calls will then fail with EBUSY. In that case just retry.
+		} while (m_work_handle.getError() == EBUSY);
```

Comment:
Yes. If there are 2 threads one of them will succeed for sure, the other one will loop and succeed next time. Similarly for N threads.